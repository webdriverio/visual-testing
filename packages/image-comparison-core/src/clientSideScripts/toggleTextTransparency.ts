/**
 * Add or remove the transparency from the text
 */
export default function toggleTextTransparency(enableTransparency: boolean): void {
    if (enableTransparency) {
        const elements = document.querySelectorAll<HTMLElement>('body *:not(script):not(style)')
        elements.forEach(el => {
            el.style.setProperty('color', 'transparent', 'important')
        })
        return
    }

    const elements = document.querySelectorAll<HTMLElement>('body [color]:not(script):not(style)')
    elements.forEach(el => {
        el.style.removeProperty('color')
    })
}
