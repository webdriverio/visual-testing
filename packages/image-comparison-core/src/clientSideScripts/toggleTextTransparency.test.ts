// @vitest-environment jsdom

import { describe, it, expect, beforeEach, afterEach } from 'vitest'
import toggleTextTransparency from './toggleTextTransparency.js'

describe('toggleTextTransparency', () => {
    beforeEach(() => {
        document.body.innerHTML = `
            <div id="testDiv">
                <p>Paragraph</p>
                <span>Span</span>
                <script>console.log('script')</script>
                <style>body { background-color: blue; }</style>
            </div>
        `
        const styleElement = document.createElement('style')
        styleElement.innerHTML = '/* head style */'
        document.head.appendChild(styleElement)
    })

    afterEach(() => {
        document.body.innerHTML = ''
        document.head.innerHTML = ''
    })

    it('should make text transparent when enabled', () => {
        toggleTextTransparency(true)

        const testDiv = document.getElementById('testDiv')

        expect(testDiv).toMatchSnapshot()
    })

    it('should remove transparent style when disabled', () => {
        toggleTextTransparency(true)
        const transparentDiv = document.getElementById('testDiv')

        expect(transparentDiv).toMatchSnapshot()

        toggleTextTransparency(false)
        const testDiv = document.getElementById('testDiv')

        expect(testDiv).toMatchSnapshot()
    })
})
