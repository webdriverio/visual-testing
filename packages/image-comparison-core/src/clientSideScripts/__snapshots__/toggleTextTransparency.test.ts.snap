// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`toggleTextTransparency > should make text transparent when enabled 1`] = `
<div
  id="testDiv"
  style="color: transparent;"
>
  
                
  <p
    style="color: transparent;"
  >
    Paragraph
  </p>
  
                
  <span
    style="color: transparent;"
  >
    Span
  </span>
  
                
  <script>
    console.log('script')
  </script>
  
                
  <style>
    body { background-color: blue; }
  </style>
  
            
</div>
`;

exports[`toggleTextTransparency > should remove transparent style when disabled 1`] = `
<div
  id="testDiv"
  style="color: transparent;"
>
  
                
  <p
    style="color: transparent;"
  >
    Paragraph
  </p>
  
                
  <span
    style="color: transparent;"
  >
    Span
  </span>
  
                
  <script>
    console.log('script')
  </script>
  
                
  <style>
    body { background-color: blue; }
  </style>
  
            
</div>
`;

exports[`toggleTextTransparency > should remove transparent style when disabled 2`] = `
<div
  id="testDiv"
  style="color: transparent;"
>
  
                
  <p
    style="color: transparent;"
  >
    Paragraph
  </p>
  
                
  <span
    style="color: transparent;"
  >
    Span
  </span>
  
                
  <script>
    console.log('script')
  </script>
  
                
  <style>
    body { background-color: blue; }
  </style>
  
            
</div>
`;
